mirrord/cli/src/connection.rs

```diff
@@ -91,6 +91,8 @@ where
             .await
             .map_err(CliError::KubernetesApiFailed)?;
 
+        k8s_api.detect_openshift(progress).await?;
```

Comment:
Create a new task for detecting openshift (progress)

1. When openshift is detected, use the warning api to leave a message about the possible issues
2. When openshift isn't detected, end the progress with "openshift not detected"
3. When you fail detecting, (i.e `detect_openshift` returns an err) use warning API to let user know check wasn't done "couldn't determine if OpenShift"